graphx/src/main/scala/org/apache/spark/graphx/api/java/JavaVertexRDDLike.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.graphx.api.java
+
+import java.util.{List => JList}
+
+import org.apache.spark.api.java.JavaRDDLike
+import org.apache.spark.api.java.function.{Function => JFunction, Function2 => JFunction2, Function3 => JFunction3}
+import org.apache.spark.graphx._
+import org.apache.spark.graphx.impl.{EdgeRDDImpl, ShippableVertexPartition}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.{Logging, Partition, TaskContext}
+
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+
+trait JavaVertexRDDLike[VD, This <: JavaVertexRDDLike[VD, This, R],
```

Comment:
I'd remove this trait and fold all of this code directly into the `JavaVertexRDD` class.  The `*RDDLike` pattern in the Java API wasn't a great design and I'd like to avoid mimicking it in new code.